site/js/modules/webui/render.js

```diff
@@ -64,6 +64,146 @@ function makeCardSection(sectionName, sectionCards) {
+// The page inputs expects a list of Input.Fields, see ./input.js for more information.
+function makePage(
+        params, context, container, onSubmitFunc,
+        page = {className: '', controlAreaHTML: '', header: '', description: '', inputs: [], buttonName: 'Submit'}) {
+    let controlAreaHTML = '';
+    if ((page.controlAreaHTML) && (page.controlAreaHTML != '')) {
+        controlAreaHTML = page.controlAreaHTML;
+    }
+
+    let headerHTML = '';
+    if ((page.header) && (page.header != '')) {
+        headerHTML = `
+            <div class="template-header secondary-color drop-shadow">
+                <h2>${page.header}</h2>
+            </div>
+        `;
+    }
+
+    let descriptionHTML = '';
+    if ((page.description) && (page.description != '')) {
+        descriptionHTML = `
+            <div class="template-description secondary-color drop-shadow">
+                <p>
+                    ${page.description}
+                </p>
+            </div>
+        `;
+    }
+
+    let infoHTML = '';
+    if ((headerHTML != '') || (descriptionHTML != '')) {
+        infoHTML = `
+            <div class="page-information">
+                ${headerHTML}
+                ${descriptionHTML}
+            </div>
+        `;
+    }
+
+    let inputSectionHTML = '';
+    if ((page.inputs) && (page.inputs.length > 0)) {
+        let inputHTML = '';
+        for (const input of page.inputs) {
+            inputHTML += input.toHTML();
+        }
+
+        inputSectionHTML = `
+            <div class="user-input-fields secondary-color drop-shadow">
+                <fieldset>
+                    ${inputHTML}
+                </fieldset>
+            </div>
+        `;
+    }
+
+    let buttonHTML = '';
+    if (onSubmitFunc) {
+        buttonHTML = `<button class="template-button">${page.buttonName}</button>`;
+    }
+
+    container.innerHTML = `
+        <div class="template-page ${page.className}">
+            <div class="template-content">
+                ${controlAreaHTML}
+                ${infoHTML}
+                ${inputSectionHTML}
+                ${buttonHTML}
+                <div class="results-area"></div>
+            </div>
+        </div>
+    `;
+
+    container.querySelector("button")?.addEventListener("click", function(event) {
+        populateResultsArea(params, context, container, page.inputs, onSubmitFunc);
+    });
+
+    container.querySelector(".user-input-fields fieldset")?.addEventListener("keydown", function(event) {
+        if (event.key != "Enter") {
+            return;
+        }
+
+        populateResultsArea(params, context, container, page.inputs, onSubmitFunc);
+    });
+
+    container.querySelectorAll(".user-input-fields fieldset input")?.forEach(function(input) {
+        input.addEventListener("blur", function(event) {
+            input.classList.add("touched");
+        });
+    });
+}
+
+function populateResultsArea(params, context, container, inputs, onSubmitFunc) {
```

Comment:
I don't like this name.
You are doing much more than just populating the results area, you are making the actual call/submit.